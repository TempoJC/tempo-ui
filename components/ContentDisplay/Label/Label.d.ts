import { LabelProps } from './types';
/**
 * Component which shows a text value as a label.
 */
declare const Label: ({ className, iconLeft, iconRight, variant, testId, color, label, uppercase, }: LabelProps) => import("react/jsx-runtime").JSX.Element;
export default Label;
