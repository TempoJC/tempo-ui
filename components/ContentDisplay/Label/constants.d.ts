export declare const LABEL_DARK = "dark";
export declare const LABEL_LIGHT = "light";
export declare const LABEL_COLOR_COCONUT = "coconut";
export declare const LABEL_COLOR_BLACKBERRY = "blackberry";
export declare const LABEL_COLOR_CHERRY = "cherry";
export declare const LABEL_COLOR_APRICOT = "apricot";
export declare const LABEL_COLOR_KIWI = "kiwi";
export declare const LABEL_COLOR_MINT = "mint";
export declare const LABEL_COLOR_WATER = "water";
export declare const LABEL_COLOR_BLUEBERRY = "blueberry";
export declare const LABEL_COLOR_GRAPE = "grape";
export declare const LABEL_COLOR_WATERMELON = "watermelon";
