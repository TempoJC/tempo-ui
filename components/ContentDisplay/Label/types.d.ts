import { ReactNode } from '../../../../node_modules/react';
import { CommonProps, VariantColor } from '../../../types';
export type LabelProps = CommonProps & {
    iconLeft?: ReactNode;
    iconRight?: ReactNode;
    label?: ReactNode;
    uppercase?: boolean;
    variant?: VariantColor;
};
