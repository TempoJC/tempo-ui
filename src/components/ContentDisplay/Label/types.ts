import type { ReactNode } from "react";
import { CommonProps, VariantColor } from "@/types";

export type LabelProps = CommonProps & {
  iconLeft?: ReactNode;
  iconRight?: ReactNode;
  label?: ReactNode;
  uppercase?: boolean;
  variant?: VariantColor;
};
