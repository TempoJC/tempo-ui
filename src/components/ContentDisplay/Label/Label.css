/* Label colors */
.tempo-label--color-coconut {
  --tempo-label-background: var(--color-coconut200);
}

.tempo-label--color-blackberry {
  --tempo-label-background: var(--color-blackberry200);
}

.tempo-label--color-cherry {
  --tempo-label-background: var(--color-cherry200);
}

.tempo-label--color-apricot {
  --tempo-label-background: var(--color-apricot200);
}

.tempo-label--color-kiwi {
  --tempo-label-background: var(--color-kiwi200);
}

.tempo-label--color-mint {
  --tempo-label-background: var(--color-mint200);
}

.tempo-label--color-water {
  --tempo-label-background: var(--color-water200);
}
.tempo-label--color-blueberry {
  --tempo-label-background: var(--color-blueberry200);
}

.tempo-label--color-grape {
  --tempo-label-background: var(--color-grape200);
}

.tempo-label--color-watermelon {
  --tempo-label-background: var(--color-watermelon200);
}

.tempo-label--light {
  --tempo-label-color: var(--color-token-light-content-inverse);
}

.tempo-label--dark {
  --tempo-label-color: var(--color-token-dark-content-inverse);
}

.tempo-label {
  display: inline-flex;
  flex-direction: row;
  height: 20px;
  padding: 0px calc(var(--size1) + 2px);
  align-self: center;
  background: var(--tempo-label-background);
}

.tempo-label__content {
  flex: none;
  align-self: center;
  padding: var(--size1) 0px;
  font: var(--typography-label-l-regular-font);
  letter-spacing: var(--typography-label-l-regular-letter-spacing);
  text-transform: var(--typography-label-l-regular-text-transform);
  color: var(--tempo-label-color);
}

.tempo-label__content--lowercase {
  text-transform: none;
}

.tempo-label--icon--left > .tempo-label__content {
  padding-left: var(--size1);
}

.tempo-label--icon--right > .tempo-label__content {
  padding-right: var(--size1);
}

.tempo-label__icon svg path {
  fill: var(--tempo-label-color);
}

.tempo-label--dark .tempo-label__icon svg path {
  fill: var(--tempo-label-color);
}
