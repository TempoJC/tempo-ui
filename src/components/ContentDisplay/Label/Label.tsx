import cn from "classnames";
import "./Label.css";
import { LabelProps } from "./types";
import {
  LABEL_COLOR_APRICOT,
  LABEL_COLOR_BLACKBERRY,
  LABEL_COLOR_BLUEBERRY,
  LABEL_COLOR_CHERRY,
  LABEL_COLOR_COCONUT,
  LABEL_COLOR_GRAPE,
  LABEL_COLOR_KIWI,
  LABEL_COLOR_MINT,
  LABEL_COLOR_WATER,
  LABEL_COLOR_WATERMELON,
  LABEL_DARK,
  LABEL_LIGHT,
} from "./constants";

/**
 * Component which shows a text value as a label.
 */
const Label = ({
  className,
  iconLeft,
  iconRight,
  variant = LABEL_COLOR_WATER,
  testId,
  color = LABEL_LIGHT,
  label,
  uppercase = true,
}: LabelProps) => (
  <div
    data-testid={testId}
    className={cn(
      "tempo-label",
      {
        "tempo-label--icon--left": iconLeft,
        "tempo-label--icon--right": iconRight,
        "tempo-label--dark": color === LABEL_DARK,
        "tempo-label--light": color === LABEL_LIGHT,
        "tempo-label--color-coconut": variant === LABEL_COLOR_COCONUT,
        "tempo-label--color-blackberry": variant === LABEL_COLOR_BLACKBERRY,
        "tempo-label--color-cherry": variant === LABEL_COLOR_CHERRY,
        "tempo-label--color-apricot": variant === LABEL_COLOR_APRICOT,
        "tempo-label--color-kiwi": variant === LABEL_COLOR_KIWI,
        "tempo-label--color-mint": variant === LABEL_COLOR_MINT,
        "tempo-label--color-water": variant === LABEL_COLOR_WATER,
        "tempo-label--color-blueberry": variant === LABEL_COLOR_BLUEBERRY,
        "tempo-label--color-grape": variant === LABEL_COLOR_GRAPE,
        "tempo-label--color-watermelon": variant === LABEL_COLOR_WATERMELON,
      },
      className,
    )}
  >
    {iconLeft && <div className="tempo-label__icon">{iconLeft}</div>}
    <div
      className={cn("tempo-label__content", {
        "tempo-label__content--lowercase": !uppercase,
      })}
    >
      {label}
    </div>
    {iconRight && <div className="tempo-label__icon">{iconRight}</div>}
  </div>
);

export default Label;
