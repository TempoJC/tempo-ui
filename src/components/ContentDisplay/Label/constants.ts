export const LABEL_DARK = "dark";
export const LABEL_LIGHT = "light";

// Label colors
export const LABEL_COLOR_COCONUT = "coconut";
export const LABEL_COLOR_BLACKBERRY = "blackberry";
export const LABEL_COLOR_CHERRY = "cherry";
export const LABEL_COLOR_APRICOT = "apricot";
export const LABEL_COLOR_KIWI = "kiwi";
export const LABEL_COLOR_MINT = "mint";
export const LABEL_COLOR_WATER = "water";
export const LABEL_COLOR_BLUEBERRY = "blueberry";
export const LABEL_COLOR_GRAPE = "grape";
export const LABEL_COLOR_WATERMELON = "watermelon";
