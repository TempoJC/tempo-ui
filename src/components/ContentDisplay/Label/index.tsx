// packages/components-content/label
export { default as Label } from "./Label";
export * from "./types";
