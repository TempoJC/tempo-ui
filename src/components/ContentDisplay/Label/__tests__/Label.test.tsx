import { render, screen } from "@testing-library/react";
import Label from "../Label";

describe("@tempo-ui/ContentDisplay/Label", () => {
  it("renders", () => {
    const { asFragment } = render(<Label label="Label" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("render icons", () => {
    const { asFragment } = render(<Label label="Label with icon" iconLeft={<div data-mock-id />} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("render with other color", () => {
    const { asFragment } = render(<Label label="Label with icon" color="dark" iconLeft={<div data-mock-id />} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("render with other variant", () => {
    const { asFragment } = render(
      <Label label="Label with icon" variant="blueberry" color="dark" iconLeft={<div data-mock-id />} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe("testId prop", () => {
    it("renders the Label with the specified data-testid attribute", () => {
      const { getByTestId } = render(<Label testId="labelID" label="Label1" />);
      const element = getByTestId("labelID");
      expect(element).toBeInTheDocument();
      expect(element).toHaveAttribute("data-testid", "labelID");
    });
  });

  it("render lowercase label", () => {
    render(<Label testId="labelLowercase" label="Label" uppercase={false} />);
    expect(screen.getByText("Label")).toHaveClass("tempo-label__content--lowercase");
  });
});
