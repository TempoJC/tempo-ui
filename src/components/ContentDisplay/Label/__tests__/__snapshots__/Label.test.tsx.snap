// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`@tempo-ui/ContentDisplay/Label render icons 1`] = `
<DocumentFragment>
  <div
    class="tempo-label tempo-label--icon--left tempo-label--light tempo-label--color-water"
  >
    <div
      class="tempo-label__icon"
    >
      <div
        data-mock-id="true"
      />
    </div>
    <div
      class="tempo-label__content"
    >
      Label with icon
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Label render with other color 1`] = `
<DocumentFragment>
  <div
    class="tempo-label tempo-label--icon--left tempo-label--dark tempo-label--color-water"
  >
    <div
      class="tempo-label__icon"
    >
      <div
        data-mock-id="true"
      />
    </div>
    <div
      class="tempo-label__content"
    >
      Label with icon
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Label render with other variant 1`] = `
<DocumentFragment>
  <div
    class="tempo-label tempo-label--icon--left tempo-label--dark tempo-label--color-blueberry"
  >
    <div
      class="tempo-label__icon"
    >
      <div
        data-mock-id="true"
      />
    </div>
    <div
      class="tempo-label__content"
    >
      Label with icon
    </div>
  </div>
</DocumentFragment>
`;

exports[`@tempo-ui/ContentDisplay/Label renders 1`] = `
<DocumentFragment>
  <div
    class="tempo-label tempo-label--light tempo-label--color-water"
  >
    <div
      class="tempo-label__content"
    >
      Label
    </div>
  </div>
</DocumentFragment>
`;
